sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -523,22 +560,45 @@ pi_result piMemBufferCreate(pi_context context, pi_mem_flags flags, size_t size,
                             void *host_ptr, pi_mem *ret_mem,
                             const pi_mem_properties *properties) {
   pi_result ret_err = PI_INVALID_OPERATION;
-  clCreateBufferWithPropertiesINTEL_fn FuncPtr = nullptr;
-
-  if (properties)
+  if (properties) {
+    clCreateBufferWithPropertiesINTEL_fn FuncPtr = nullptr;
+    const size_t propSize = sizeof(properties) / sizeof(pi_mem_properties);
     // First we need to look up the function pointer
     ret_err = getExtFuncFromContext<clCreateBufferWithPropertiesName,
                                     clCreateBufferWithPropertiesINTEL_fn>(
         context, &FuncPtr);
+    if (FuncPtr) {
+      std::vector<pi_mem_properties> supported(properties,
+                                               properties + propSize);
+      // Go through buffer properties. If there is one, that shall be propagated
+      // to an OpenCL runtime - check if this property is being supported.
+      for (const auto &prop = supported.begin(); prop != supported.end();
+           ++(*prop)) {
+        if (SupportedExtensions.find(context) == SupportedExtensions.end())
+          ret_err = getSupportedExtensionsWithinContext(context);
+        // Check if PI_MEM_PROPERTIES_CHANNEL property is supported. If it's
+        // not - just ignore it, as it's an optimization hint.
+        if (*prop == PI_MEM_PROPERTIES_CHANNEL) {
```

Comment:
Could you please clarify why these checks are needed, can we just pass properties as is so if some of them is not supported we will get an error?

Reply:
From the spec:
On some targets manual assignment of buffers to memory regions can improve memory bandwidth.  This extension adds a buffer property to indicate in which memory channel a particular buffer should be allocated.  _This information is an optimization hint to the runtime and thus it is legal to ignore_.